moto/ec2/responses/subnets.py

```diff
@@ -11,9 +11,7 @@ def create_subnet(self) -> str:
         ipv6_cidr_block = self._get_param("Ipv6CidrBlock")
         availability_zone = self._get_param("AvailabilityZone")
         availability_zone_id = self._get_param("AvailabilityZoneId")
-        tags = self._get_multi_param("TagSpecification")
-        if tags:
-            tags = tags[0].get("Tag")
+        tags = self._get_multi_param("TagSpecification", skip_result_conversion=True)
```

Comment:
```suggestion
        tags = self._parse_tag_specification()
```

Tag-handling is.. how shall I put this delicately.. a bit of a mess. Across all services.

But for EC2, I think that the `parse_tag_specification` is the best approach. 

If you wouldn't mind refactoring that method to add this validation functionality about missing/incorrect `ResourceType`s, it will be really easy to extend this behaviour to other resource-types in the future.